From 6c69aca68d96b933f8d36c0fa2117ef6878de951 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:59:02 +0200
Subject: [PATCH 35/50] AnnaThermostat: add outdoor_temperature measurement

Map outdoor_temperature to a Domoticz temperature sensor in degrees Celsius.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 0588a98de..5b1923c0c 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1391,6 +1391,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendTempSensor(applianceIDhex + 5, batterypercentage, temperature, ApplianceName + " " + sname);;
                 }
             }
+			else if (sname == "outdoor_temperature")
+			{
+				if (!tmpstr.empty())
+				{
+					float temperature = (float)atof(tmpstr.c_str());
+					SendTempSensor(applianceIDhex + 6, batterypercentage, temperature, ApplianceName + " " + sname);
+				}
+			}
 			else if (sname == "domestic_hot_water_setpoint")
 			{
 				if (!tmpstr.empty())
-- 
2.51.0

